adhocteam/Head-Start-TTADP
# CD: Seed db in lower envs, reset sandbox each deploy

**Description of change**
Defines three pathways for database:
- sandbox db: undo seed, undo migrations, run migrations, run seed
- dev db: undo seed, run migrations, run seed
- staging and prod dbs: just run migrations

I tried dropping the database completely on every deploy to sandbox (this felt cleaner to me) but this is hard to pull off from within cf.  It wouldn't let me recreate the db after I drop it. I'm sure there's a way to do it, but in the interest of time, I thought undoing the seed and migrations were enough 99% of the time, unless we've really screwed up the db. 

**How to test**
- after merging this PR to the feature branch check the sandbox log stream, you should see all the db tasks run successfully (undo seed, undo migrate, migrate, seed). I verified the database transformations worked on my last deploy to sandbox.

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/52

**Checklist**

- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] Documentation updated - I didn't add any new docs here, but there's a description on the new "seed and migrate db" command, let me know if you think the code isn't self-documenting here
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
